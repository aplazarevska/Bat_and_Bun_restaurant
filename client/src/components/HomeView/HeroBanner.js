import React from 'react';

const HeroBanner = ({ imgUrl }) => {
  return (
    <div data-testid="hero-image" id="hero-image" style={{ backgroundImage: `url(${imgUrl})` }}>
    </div>
  )
}

export default HeroBanner;
