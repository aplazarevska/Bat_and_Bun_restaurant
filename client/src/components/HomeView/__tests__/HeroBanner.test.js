import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import HeroBanner from '../HeroBanner';
import { afterEach, expect } from '@jest/globals';

const imgUrl = "./img/hero-image.jpg";

afterEach(() => {
  cleanup();
});

test("Should render hero banner component that has a background image", () => {
  render(<HeroBanner imgUrl={imgUrl} />);
  const heroBannerElement = screen.getByTestId('hero-image');
  expect(heroBannerElement).toBeInTheDocument();
  expect(heroBannerElement).toHaveStyle(`background-image: url(${imgUrl})`);
});

test("Matches snapshot", () => {
  const tree = renderer.create(<HeroBanner imgUrl={imgUrl} />).toJSON();
  expect(tree).toMatchSnapshot();
});
