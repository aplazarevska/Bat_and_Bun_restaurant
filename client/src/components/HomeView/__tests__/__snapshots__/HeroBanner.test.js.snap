// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  data-testid="hero-image"
  id="hero-image"
  style={
    Object {
      "backgroundImage": "url(./img/hero-image.jpg)",
    }
  }
/>
`;
